cli/pcluster/utils.py

```diff
@@ -166,11 +166,18 @@ def get_supported_features(region, feature):
     try:
         features = _get_json_from_s3(region, "features/feature_whitelist.json")
         supported_features = features.get("Features").get(feature)
-    except (ValueError, ClientError, KeyError):
-        print(
-            "Failed validate %s. This is probably a bug on our end. Please set sanity_check = false and retry" % feature
-        )
-        exit(1)
+    except (ValueError, ClientError, KeyError) as e:
+        code = e.response.get("Error").get("Code")
```

Comment:
I don't think all the exceptions we are catching here have the Error.Code and Error.Message fields

Reply:
Ok, I've switched this to check only when it's `ClientError`. For example, with invalid credentials this results in:

```bash
$ pcluster create efa-test
Beginning cluster creation for cluster: efa-test
The AWS Access Key Id you provided does not exist in our records.
```
When I change the file to a non-existant file, for example `features/feature_whitelist2.json`
```bash
$ pcluster create efa-test
Beginning cluster creation for cluster: efa-test
Failed validate efa. This is probably a bug on our end. Please submit an issue https://github.com/aws/aws-parallelcluster/issues/new/choose
```